Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,676 @@
+# -*- coding: utf-8 -*-
+# Written by Egbert Bouman, Mihai Capotă, Elric Milon, and Ardhi Putra Pratama H
+"""Manage boosting of swarms"""
+import ConfigParser
+import json
+import logging
+import os
+import random
+from binascii import hexlify, unhexlify
+
+import errno
+import libtorrent as lt
+
+from Tribler.Core.DownloadConfig import DownloadStartupConfig
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.Core.Utilities.install_dir import determine_install_dir
+from Tribler.Core.simpledefs import DLSTATUS_SEEDING, NTFY_TORRENTS, NTFY_UPDATE, NTFY_CHANNELCAST
+from Tribler.Main.globals import DefaultDownloadStartupConfig
+from Tribler.Policies.BoostingSource import ChannelSource
+from Tribler.Policies.BoostingSource import DirectorySource
+from Tribler.Policies.BoostingSource import RSSFeedSource
+from Tribler.dispersy.taskmanager import TaskManager
+
+NUMBER_TYPES = (int, long, float)
+
+# CONFIG_FILE = "boosting.ini"
+TRIBLER_ROOT = determine_install_dir()
+CONFIG_FILE = os.path.join(TRIBLER_ROOT, "boosting.ini")
+
+
+def lev(trnt_1, trnt_2):
+    """
+    Calculates the Levenshtein distance between a and b.
+    """
+    trnt_1_len, trnt_2_len = len(trnt_1), len(trnt_2)
+    if trnt_1_len > trnt_2_len:
+        # Make sure trnt_1_len <= trnt_2_len, to use O(min(trnt_1_len,trnt_2_len)) space
+        trnt_1, trnt_2 = trnt_2, trnt_1
+        trnt_1_len, trnt_2_len = trnt_2_len, trnt_1_len
+
+    current = range(trnt_1_len + 1)
+    for i in range(1, trnt_2_len + 1):
+        previous, current = current, [i] + [0] * trnt_1_len
+        for j in range(1, trnt_1_len + 1):
+            add, delete = previous[j] + 1, current[j - 1] + 1
+            change = previous[j - 1]
+            if trnt_1[j - 1] != trnt_2[i - 1]:
+                change += 1
+            current[j] = min(add, delete, change)
+
+    return current[trnt_1_len]
+
+def source_to_string(source_obj):
+    return hexlify(source_obj) if len(source_obj) == 20 and not (source_obj.startswith('http://')
+                                                                 or source_obj.startswith('https://')) else source_obj
+
+def string_to_source(source_str):
+    return source_str.decode('hex') \
+        if len(source_str) == 40 and not (os.path.isdir(source_str) or source_str.startswith('http://')) else source_str
+
+class BoostingPolicy(object):
+    """
+    Base class for determining what swarm selection policy will be applied
+    """
+
+    def __init__(self, session):
+        self.session = session
+        self.key = lambda x: None
+        # function that checks if key can be applied to torrent
+        self.key_check = lambda x: None
```

Comment:
I think the `key_check` lambda function should always return a bool object so I would rather return `False`.